build/components/setup.tex

```diff
@@ -64,17 +64,27 @@
 
 % AUTISTIC CONDITIONALS
 % This hides ToF and ToT if they are empty
-\makeatletter
-\AtEndEnvironment	{figure}{\gdef\there@is@a@figure{}}
-\AtEndEnvironment	{table}{\gdef\there@is@a@table{}}
-
-\AtEndDocument		{\ifdefined\there@is@a@figure\label{fig:was:used:in:doc}\fi}
-\AtEndDocument		{\ifdefined\there@is@a@table\label{tab:was:used:in:doc}\fi}
-\newcommand{\conditionalLoF}{\@ifundefined{r@fig:was:used:in:doc}{}
-    {\addcontentsline{toc}{section}{\listfigurename} \listoffigures \cleardoublepage}}%
-\newcommand{\conditionalLoT}{\@ifundefined{r@tab:was:used:in:doc}{}
-    {\addcontentsline{toc}{section}{\listtablename} \listoftables \cleardoublepage}}%
-\makeatother
+\newtotcounter{figCount}
+\newtotcounter{tabCount}
+\let\oldTab=\table
+\let\oldFig=\figure
+\renewcommand{\figure}{\stepcounter{figCount}\oldFig}
+\renewcommand{\table}{\stepcounter{tabCount}\oldTab}

```

Comment:
funktionierende Zwischenlösung
```suggestion
\let\oldTabTOC=\table
\let\oldFigTOC=\figure
\renewcommand{\figure}{\stepcounter{figCount}\oldFigTOC}
\renewcommand{\table}{\stepcounter{tabCount}\oldTabTOC}

```

Reply:
das ganze ist ziemlich unordentlich... Ich bitte im Nachgang um ein Aufräumen der setups^^

Reply:
Löst das die Endlosschleife?

Reply:
ist wohl Augenscheinlich keine Endlosschleife sondern eher in Versuch von Latex eine variable doppelt zu erstellen wo es sich dann verabschieded und endloss weiterdenkt was es machen soll.
_ist alles weniger latex mehr wissen wo was passiert hehe... deswegen will ich das definitv aufgeräumt haben, aktuell kann man sich da echt schlecht zurechtfinden wenn man nicht weis wo man suchen muss_

Reply:
evtl ist auch eine technische doku angebracht :D

Reply:
Der Log sagte mir was anderes, ich hab mal versucht zu debuggen, ich schick mal einen Auszug der 10 Millionen Zeilen rein, moment.

Reply:
```
\cl@tabCount ->
{\endgroup}

\oldTab ->\stepcounter {tabCount}\oldTab 

\stepcounter #1->\addtocounter {#1}\@ne \begingroup \let \@elt \@stpelt \csname
 cl@#1\endcsname \endgroup 
#1<-tabCount

\addtocounter #1#2->\@ifundefined {c@#1}{\@nocounterr {#1}}{\global \advance \c
sname c@#1\endcsname #2\relax }
#1<-tabCount
#2<-\@ne 

\@ifundefined #1->\ifcsname #1\endcsname \@ifundefin@d@i \else \@ifundefin@d@ii
 \fi {#1}
#1<-c@tabCount

\@ifundefin@d@i #1\fi #2->\fi \expandafter \ifx \csname #2\endcsname \relax \@i
fundefin@d@ii \fi \@secondoftwo 
#1<-\else \@ifundefin@d@ii 
#2<-c@tabCount

\@secondoftwo #1#2->#2
#1<-\@nocounterr {tabCount}
#2<-\global \advance \csname c@tabCount\endcsname \@ne \relax 
{\global}
{\relax}
{\begingroup}
{\let}
```
Und das wiederholt er so lange bis ich das ganze beende. Nur werde ich daraus nicht schlau. Aber wenn du das mit oben gefixt hast, passt das ja.

Reply:
Ja latex ist leicht verwirrend und undurchsichtig (⌒_⌒;) 
deswegen am besten immer nur die logs anschauen die warnings oder errros sind. 
Ich hab auch nur die stelle gefunden da es zuvor noch kompiliert hatte und die neuesten Änderungen es dann gekillt hatten.